### PR TITLE
Onboarding skills repo

### PR DESCRIPTION
**Pull request by zane-neo**
### Description
Onboarding skills repo

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/4378

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: manifests/2.12.0/opensearch-2.12.0.yml [hunk 1]
```diff
@@ -233,3 +233,17 @@ components:
...
       - common-utils
+  - name: skills
+    repository: https://github.com/opensearch-project/skills.git
+    ref: 2.x
+    platforms:
+      - linux
+      - windows
+    checks:
+      - gradle:properties:version
+      - gradle:dependencies:opensearch.version
```
Comment (peterzhuamazon) [inline manifests/2.12.0/opensearch-2.12.0.yml:244]: Please remove this as we dont need this section anymore.

File: manifests/2.12.0/opensearch-2.12.0-concurrent-search-test.yml [hunk 2]
```diff
@@ -198,3 +198,12 @@ components:
         opensearch.experimental.feature.concurrent_segment_search.enabled: true
         search.concurrent_segment_search.enabled: true
         search.concurrent.max_slice_count: 2
+  - name: skills
```
Comment (peterzhuamazon) [inline manifests/2.12.0/opensearch-2.12.0-concurrent-search-test.yml:201]: Please add a newline before this, and confirm is skills GA or experimental?

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]

**Comment by peterzhuamazon** [comment]
We also need to wait for a bit before merging, as skills repo has wrong rootname, wrong m2 folder, and others.
Zan is working to fix those.

**Review by prudhvigodithi — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]
Thanks @zane-neo.

I also tweaked a few issues I see in the workflow ymls, and resolve conflicts.

Will ask @prudhvigodithi to give another approval. Thanks.

**Comment by prudhvigodithi** [comment]
Hey @zane-neo can you please take care of this issue https://github.com/opensearch-project/skills/issues/188#issue-2121422972 ?
Thank you

**Pull request merged**